### PR TITLE
HDDS-5879. [Ozone-Streaming] OzoneBucket add the createMultipartStreamKey method

### PR DESCRIPTION
**Pull request by guohao-rosicky**
https://issues.apache.org/jira/browse/HDDS-5879

OzoneBucket and RpcClient not support multipart upload with Stream.
Add the createMultipartStreamKey method

### CODE REVIEW

**Comment by captainzmc** [comment]
Thanks @guohao-rosicky for your contribution. Could you update the URL of jira in comment?  Also, the current code check style still has some problems, and it is better to add some UT to test the new interface.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java [hunk 1]
```diff
@@ -730,6 +730,23 @@ public OzoneOutputStream createMultipartKey(String key, long size,
...
+   * @param size
+   * @param partNumber
+   * @param uploadID
+   * @param replicationConfig
+   * @return OzoneDataStreamOutput
+   * @throws IOException
+   */
+  public OzoneDataStreamOutput createMultipartStreamKey(String key,
+      long size, int partNumber, String uploadID,
+      ReplicationConfig replicationConfig) throws IOException {
```
[the remark this answers is not in the document]
Comment (guohao-rosicky) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java:745]: @szetszwo 
I noticed it. Thanks for reminding me. I have revised it.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 2]
```diff
@@ -1133,6 +1133,68 @@ public OzoneOutputStream createMultipartKey(String volumeName,
...
+    keyOutputStream
+        .addPreallocateBlocks(
+            openKey.getKeyInfo().getLatestVersionLocations(),
+            openKey.getOpenVersion());
+
+    FileEncryptionInfo feInfo = openKey.getKeyInfo().getFileEncryptionInfo();
+    if (feInfo != null) {
+      // todo: need to support file encrypt,
+      //  https://issues.apache.org/jira/browse/HDDS-5892
+      return null;
```
Comment (szetszwo) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:1192]: Let's throw a UnsupportedOperationException.
```
      throw new UnsupportedOperationException(
          "FileEncryptionInfo is not yet supported in createMultipartStreamKey");
```

**Review by szetszwo — COMMENTED** [review]
@guohao-rosicky , thanks for the update.

The new testMultiPartUploadWithStream() failed.   Please take a look.

I also have a minor comment inlined.

**Comment by szetszwo** [comment]
@guohao-rosicky , TestOzoneRpcClientAbstract.testMultiPartUploadWithStream failed again in the last build.  Please take a look.

**Comment by guohao-rosicky** [comment]
> @guohao-rosicky , TestOzoneRpcClientAbstract.testMultiPartUploadWithStream failed again in the last build. Please take a look.

Thanks @szetszwo , please TBR.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**